src/zenml/integrations/modal/flavors/modal_orchestrator_flavor.py

```diff
@@ -0,0 +1,169 @@
+#  Copyright (c) ZenML GmbH 2025. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Modal orchestrator flavor."""
+
+from enum import Enum
+from typing import TYPE_CHECKING, Optional, Type
+
+from zenml.config.base_settings import BaseSettings
+from zenml.orchestrators import BaseOrchestratorConfig, BaseOrchestratorFlavor
+from zenml.utils.secret_utils import SecretField
+
+if TYPE_CHECKING:
+    from zenml.integrations.modal.orchestrators import ModalOrchestrator
+
+from zenml.integrations.modal import MODAL_ORCHESTRATOR_FLAVOR
+
+
+class ModalExecutionMode(str, Enum):
+    """Execution modes for the Modal orchestrator.
+
+    Attributes:
+        PIPELINE: Execute entire pipeline in one Modal sandbox (fastest, default).
+        PER_STEP: Execute each step in a separate Modal sandbox (granular control,
+            better for debugging, allows step-specific resources).
+    """
+
+    PIPELINE = "pipeline"
+    PER_STEP = "per_step"
+
+
+class ModalOrchestratorSettings(BaseSettings):
+    """Modal orchestrator settings.
+
+    Attributes:
+        gpu: The type of GPU to use for the pipeline execution (e.g., "T4",
+            "A100"). Use ResourceSettings.gpu_count to specify the number of GPUs.
+        region: The region to use for the pipeline execution.
+        cloud: The cloud provider to use for the pipeline execution.
+        modal_environment: The Modal environment to use for the pipeline execution.
+        timeout: Maximum execution time in seconds (default 24h).
+        execution_mode: Execution mode - PIPELINE (fastest) or PER_STEP (granular).
+        max_parallelism: Maximum number of parallel sandboxes (for PER_STEP mode).
+        synchronous: Wait for completion (True) or fire-and-forget (False).
+    """
+
```

Comment:
I think having an option to specify the sandbox name would be nice?

Reply:
done